client/src/util/typedCrossfilter/sort.js

```diff
@@ -12,6 +36,18 @@ function insertionsort(a, lo, hi) {
   return a;
 }
 
+function insertionsortFloats(a, lo, hi) {
+  for (let i = lo + 1; i < hi + 1; i += 1) {
+    const x = a[i];
+    let j;
+    for (j = i; j > lo && gt(a[j - 1], x); j -= 1) {
```

Comment:
gt/lt only works on Number, as it calls Number.isNaN.   So it will throw an error for other types of arrays (ie, those that are not Float32/Float64).

we could create two versions of gt/lt, one that is for floats and one that does the default </>, but that actually seemed more opaque to me.

So it wasn't done for perf, but for comparison semantics (only apply the NaN test for floats)

Reply:
Actually, I don't think that's true. Looks like the docs for Number.isNaN `true if the given value is NaN and its type is Number; otherwise, false` [isNaN Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN)

I was thinking that we could just use gt()/lt() everywhere to simplify code paths if it's not a performance issue

Reply:
filed as #731 for future (possible) refactoring project

Reply:
👍